src/DotNetty.Buffers/ByteBufferUtil.cs

```diff
@@ -569,5 +569,12 @@ static void AppendHexDumpRowPrefix(StringBuilder dump, int row, int rowStartInde
                 dump.Append('|');
             }
         }
+
+        public static string DecodeString(IByteBuffer src, int readerIndex, int len, Encoding encoding)
+        {
+            if (len == 0) return string.Empty;
+            
+            return encoding.GetString(src.Array.Slice(len));
```

Comment:
@nayato Sorry, i didn't understand 'CompositeByteBuffer'. But ToString of AbstractByteBuf is expansive method. I think that ToString CompositeByteBuffer is special case, so this method will be overrided. But Netty(Java) don't implemented ToString of CompositeByteBuffer. So i think that this implementation is TBD.
And according to reference Netty(Java), I think that HasArray check in ToString should be removed.